### PR TITLE
Pptp 2813 content changes

### PR DESCRIPTION
**Pull request by mauro-fanton**
### Description of Work carried through

Brief description of what/why/how.

### CODE REVIEW

File: app/config/FrontendAppConfig.scala [hunk 1]
```diff
@@ -162,5 +164,5 @@ class FrontendAppConfig @Inject() (
   def addMemberToGroupUrl: String =
     configuration.get[String]("urls.addMemberToGroup")
 
-
+  val taxRegimeStartDate: LocalDate = LocalDate.of(2023, 4, 1)
```
Comment (lmcewanhmrc) [inline app/config/FrontendAppConfig.scala:167]: kinda sounds like start of the PPT itself, needs a better name I think 🤔

File: app/config/FrontendAppConfig.scala [hunk 2]
```diff
@@ -163,4 +167,22 @@ class FrontendAppConfig @Inject() (
...
+    */
+  def overrideSystemDateTime: Option[LocalDateTime] = {
+    Try(configuration
+      .getOptional[String]("features.override-system-date-time")
+      .map(LocalDateTime.parse(_, DateTimeFormatter.ISO_LOCAL_DATE_TIME)))
+    match {
+      case Success(date) => date
+      case Failure(_) => None
+    }
+  }
```
Comment (lmcewanhmrc) [inline app/config/FrontendAppConfig.scala:190]: Should we move some of this elsewhere? Kinda a bit more than just config...

File: app/controllers/returns/credits/ConfirmPackagingCreditController.scala [hunk 3]
```diff
@@ -55,13 +58,15 @@ class ConfirmPackagingCreditController @Inject()(
 
 
   private def displayView(creditBalance: CreditBalance, mode: Mode)(implicit request: DataRequest[_]): Result = {
+    val isOnOrAfterApril2023 = !midnight1stApril2023.isAfter(edgeOfSystem.localDateTimeNow)
```
Comment (lmcewanhmrc) [inline app/controllers/returns/credits/ConfirmPackagingCreditController.scala:61]: 🤔

File: conf/application.conf [hunk 4]
```diff
@@ -172,6 +172,7 @@ features {
     deRegistrationEnabled = true
     amendsEnabled = true
     changeOfGroupLead = true
+    override-system-date-time = "20"
```
Comment (lmcewanhmrc) [inline conf/application.conf:175]: Maybe 'false'... ie set a better example 😆 ?

File: test/config/FrontendAppConfigSpec.scala [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    "return None if override not present" in {
+      when(configuration.getOptional[String](any)(any)) thenReturn None
+      frontendAppConfig.overrideSystemDateTime mustBe empty
+    }
+
+  }
+
+  private def validAppConfig(overrideSystemDateTime: String): Config =
```
Comment (lmcewanhmrc) [inline test/config/FrontendAppConfigSpec.scala:54]: Doesn't look like we need most of this?

File: test/config/FrontendAppConfigSpec.scala [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+import org.mockito.integrations.scalatest.ResetMocksAfterEachTest
+import org.scalatest.BeforeAndAfterEach
+import com.typesafe.config.{Config, ConfigFactory}
+import org.scalatestplus.play.PlaySpec
+import play.api.Configuration
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+
+import java.time.LocalDateTime
+
+class FrontendAppConfigSpec extends PlaySpec
```
Comment (lmcewanhmrc) [inline test/config/FrontendAppConfigSpec.scala:30]: Merge made a bit of a muck of this. Tidy up?

File: test/util/EdgeOfSystemSpec.scala [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+      Numeric.Implicits
+    }
+    
+    "use the override time if it's there" in {
+      val exampleDateTime = LocalDateTime.of(2023, 4, 1, 12, 11, 10)
+      when(frontendAppConfig.overrideSystemDateTime) thenReturn Some(exampleDateTime)
+      edgeOfSystem.localDateTimeNow mustBe exampleDateTime
+    }
+    
+    "use the system time if override not present" ignore { // todo
```
Comment (lmcewanhmrc) [inline test/util/EdgeOfSystemSpec.scala:48]: What to do here?

File: conf/application.conf (hunk 4 above)
Comment (mauro-fanton) [reply conf/application.conf:175]: yes, I would put false there

File: test/util/EdgeOfSystemSpec.scala [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    "use the override time if it's there" in {
+      val exampleDateTime = LocalDateTime.of(2023, 4, 1, 12, 11, 10)
+      when(frontendAppConfig.overrideSystemDateTime) thenReturn Some(exampleDateTime.toString)
+      edgeOfSystem.localDateTimeNow mustBe exampleDateTime
+    }
+    
+    "use the system time if override not present" in {
+      when(frontendAppConfig.overrideSystemDateTime) thenReturn None
+
+      edgeOfSystem.localDateTimeNow.truncatedTo(ChronoUnit.MINUTES) mustBe
```
Comment (lmcewanhmrc) [inline test/util/EdgeOfSystemSpec.scala:48]: Nice! 🏆

**Review by lmcewanhmrc — APPROVED** [review]

**Pull request merged**